CONTRIBUTING.md

```diff
@@ -31,10 +31,135 @@ Please be aware of the following things when filing bug reports.
 
 If the bug report is missing this information then it'll take us longer to fix the issue. We will probably ask for clarification, and barring that if no response was given then the issue will be closed.
 
+## Writing a Pull Request
+Submitting a pull request is fairly simple, just make sure it focuses on a single aspect and doesn't manage to have scope creep and it's probably good to go.
 
-## Submitting a Pull Request
+Most attributes and data structures are broken up in to a file for each related class. For example, `disnake.Guild` is defined in [disnake/guild.py](disnake/guild.py), and `disnake.GuildPreview` is defined in [disnake/guild_preview.py](disnake/guild_preview.py). For example, writing a new feature to `disnake.Guild` would go in [disnake/guild.py](disnake/guild.py), as part of the `disnake.Guild` class.
 
-Submitting a pull request is fairly simple, just make sure it focuses on a single aspect and doesn't manage to have scope creep and it's probably good to go. It would be incredibly lovely if the style is consistent to that found in the project. This project follows PEP-8 guidelines (mostly) with a column limit of 100 characters.
+### Adding a new API Feature
+
+However, adding a new feature that interfaces with the API requires also updating the [disnake/types](disnake/types) directory. We request that when making or receiving payloads from the API, they are typed and the typehints are used on the functions that are processing said data. As an example, we'll take a look at ``disnake.abc.Messageable.pins`` (defined in [disnake/abc.py](disnake/abc.py)).
+
+```py
+    async def pins(self) -> List[Message]:
+        channel = await self._get_channel()
+        state = self._state
+        data = await state.http.pins_from(channel.id)
+        return [state.create_message(channel=channel, data=m) for m in data]
+```
+*docstring removed for breverity*
+
+Here we have several things occuring. First, we have typehinted the return type of this method to return a list of Messages. As disnake supports python 3.8, we must use typing imports instead of subscripting built-ins--hence the capital ``List``.
```

Comment:
```suggestion
Here we have several things occuring. First, we have typehinted the return type of this method to return a list of Messages. As disnake supports python 3.8, we must use typing imports instead of subscripting built-ins — hence the capital ``List``.
```
emdash

Reply:
https://github.com/DisnakeDev/disnake/pull/601/commits/b287c7c5404b798ec2598c6ae38c207fe129fa7c